plugins/modules/fabric_devices_workflow_manager.py

```diff
@@ -0,0 +1,4080 @@
+        self.status = "success"
+        return self
+
+    def get_want(self, config):
+        """
+        Get the SDA fabric devices related information from playbook.
+
+        Parameters:
+            config (list of dict): Playbook details contains the details of fabric devices.
+        Returns:
+            self: The current object with updated fabric devices details.
+        Description:
+
+        """
+
+        fabric_devices = config.get("fabric_devices")
+        if not fabric_devices:
+            self.msg = "The parameter 'fabric_devices' is missing under the 'config'."
+            self.status = "failed"
+            return self
+
+        self.get_want_fabric_devices(fabric_devices).check_return_status()
+
+        self.log("Desired State (want): {requested_state}".format(requested_state=self.want), "INFO")
+        self.msg = "Successfully retrieved details from the playbook."
+        self.status = "success"
+        return self
+
+    def update_l2_handoff(self, have_l2_handoff, want_l2_handoff,
+                          device_ip, result_fabric_device_response,
+                          result_fabric_device_msg):
+        """
+        Create L2 Handoff in a fabric device with fields provided in playbook.
+
+        Parameters:
+            have_l2_handoff (list): L2 Handoff details from the Cisco Catalyst Center.
+            want_l2_handoff (dict): L2 Handoff details from the user playbook.
+            device_ip (str): The IP of the fabric device.
+            result_fabric_device_response (dict): The response status of the fabric device.
+            result_fabric_device_msg (dict): The message status of the fabric device.
+
+        Returns:
+            self (object): The current object with added L2 Handoff information.
+        Description:
+
+        """
+
+        create_l2_handoff = []
+        l2_handoff_index = -1
+
+        # The L2 Handoffs doesnot support updation
+        # So find which L2 handoffs need to be created and which needs to be ignored
+        for item in want_l2_handoff:
+            l2_handoff_index += 1
+            if not (have_l2_handoff and have_l2_handoff[l2_handoff_index]):
+                create_l2_handoff.append(item)
+
+        try:
+            payload = {"payload": create_l2_handoff}
+            task_name = "add_fabric_devices_layer2_handoffs"
+            task_id = self.get_taskid_post_api_call("sda", task_name, payload)
+            if not task_id:
+                self.msg = (
+                    "Unable to retrive the task_id for the task '{task_name}'."
+                    .format(task_name=task_name)
+                )
+                self.set_operation_result("failed", False, self.msg, "ERROR")
+                return self
+
+            success_msg = (
+                "L2 Handoff(s) '{l2_handoff}' added successfully in the device with IP '{ip}."
+                .format(l2_handoff=create_l2_handoff, ip=device_ip)
+            )
```

Comment:
self.log("Starting task status check for task ID: {task_id}".format(task_id=task_id), "DEBUG")